GraduationAPI/PokemonAPI2.0/PokemonAPI2.0/Interfaces/IPokemonRepository.cs

```diff
@@ -0,0 +1,20 @@
+﻿using PokemonWEB.Models;
+using PokemonWEB.Models.Action;
+
+namespace PokemonWEB.Interfaces;
+
+public interface IPokemonRepository
+{
+    Pokemon GetPokemon(Guid Id);
+    ICollection<Pokemon> GetPokemons(int countPokemon);
+
+    void CreatePokemon(Guid ownerId, Guid categoryId, Pokemon pokemon);
+    bool UpdatePokemon(Guid ownerId, Guid categoryId, Pokemon pokemon);
+    bool DeletePokemon(Pokemon pokemon);
+    void SubstractDamage(Pokemon pokemon, int damage);
+    bool IsDefead(Pokemon pokemon);
+    void PerfomAttack(Pokemon selfPokemon, Pokemon opponentPokemon);
+    void CaluclateDamage(Pokemon selfPokemon, Ability ability, Pokemon opponentPokemon);
+    bool PokemonExists(Guid Id);
```

Comment:
If you really need repository, you should not have any business logic in repo, only crud and relationships management